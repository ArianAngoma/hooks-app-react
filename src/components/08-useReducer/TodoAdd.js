import React from "react";
import useForm from "../../hooks/useForm";

const TodoAdd = React.memo(({handleAddTodo}) => {
    console.log('TodoAdd de nuevo D:')

    const [{description}, handleInputChange, reset] = useForm({
        description: ''
    });

    const handleSubmit = (e) => {
        e.preventDefault();

        if (description.trim().length > 0) {
            handleAddTodo({
                id: new Date().getTime(),
                description: description,
                done: false
            });
            reset();
        }
    }

    return (
        <>
            <h4>Agregar Todo</h4>
            <hr/>

            <form onSubmit={handleSubmit}>
                <input type='text'
                       name='description'
                       className='form-control'
                       placeholder='Tarea'
                       autoComplete='off'
                       onChange={handleInputChange}
                       value={description}/>

                <button className='btn btn-outline-primary mt-1 w-100'
                        type='submit'>
                    Agregar
                </button>
            </form>
        </>
    )
})

export default TodoAdd;