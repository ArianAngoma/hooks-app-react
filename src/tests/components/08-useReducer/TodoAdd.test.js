import {mount, shallow} from "enzyme";
import TodoAdd from "../../../components/08-useReducer/TodoAdd";

describe('Test on <TodoAdd/>', () => {
    const handleAddTodo = jest.fn();

    const wrapper = mount(
        <TodoAdd handleAddTodo={handleAddTodo}/>
    );

    test('Should display correctly', () => {
        expect(wrapper).toMatchSnapshot();
    });

    test('Should NOT call handleAddTodo', () => {
        const formSubmit = wrapper.find('form').prop('onSubmit');
        formSubmit({
            preventDefault() {
            }
        });

        expect(handleAddTodo).toHaveBeenCalledTimes(0);
    });

    test('Should call the function handleAddTodo', () => {
        const value = 'Aprender React';

        wrapper.find('input').simulate('change', {
            target: {
                value,
                name: 'description'
            }
        });

        const formSubmit = wrapper.find('form').prop('onSubmit');
        formSubmit({
            preventDefault() {
            }
        });

        expect(handleAddTodo).toHaveBeenCalledTimes(1);
        expect(handleAddTodo).toHaveBeenCalledWith(expect.any(Object));
        expect(handleAddTodo).toHaveBeenCalledWith({
            id: expect.any(Number),
            description: value,
            done: false
        });
    });
})