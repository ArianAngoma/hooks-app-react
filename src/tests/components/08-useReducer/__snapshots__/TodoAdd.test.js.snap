// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test on <TodoAdd/> Should display correctly 1`] = `
Array [
  <h4>
    Agregar Todo
  </h4>,
  <hr />,
  <form
    onSubmit={[Function]}
  >
    <input
      autoComplete="off"
      className="form-control"
      name="description"
      onChange={[Function]}
      placeholder="Tarea"
      type="text"
      value=""
    />
    <button
      className="btn btn-outline-primary mt-1 w-100"
      type="submit"
    >
      Agregar
    </button>
  </form>,
]
`;
